specs/unified_kernel_image.md

```diff
@@ -159,3 +159,22 @@ that tools staging or consuming UKIs have a common place to store and look for t
 
 The installed UKIs should have a filename `<version format specification>.efi`, i.e. the filename is left to
 implementers but must be valid for comparisons according to the [Version Format Specification](version_format_specification.md).
+
+UKI addons (cmdline PE addons, systemd-sysext extensions for the initrd) built centrally by distributions and
+installed via package manager should be installed into locations depending on whether they are global UKI addons,
+that should be applied to all UKIs installed in the ESP, or whether they should only be applied to a
+single specific UKI.
+
+Global UKI addons should be installed into `/usr/lib/modules/uki.extra.d/`. UKI-specific addons should be installed
+to `/usr/lib/modules/$UNAME/$UKI.efi.extra.d/`, where `$UNAME` is the output of `uname -r` of the kernel included
+in the UKI and `$UKI` is the name of the corresponding centrally built UKI with the `.efi` extension stripped.
+
+The installed UKI addons should have a filename `<version format specification>.addon.efi`, i.e. the filename
```

Comment:
this might be misread as if addons should not carry identifying names, just version numbers. But I'd assume the opposite: that they more likely have identifying names, and less likely version ids.

i.e. let's say we start passing the machine ID as an addon into the booted system (so that the initrd can start using it, before transitioning into the host). In that case the addon should probably just be called "machine-id.addon.efi", without any version, since the version is never going to change.